### DISCUSSION TITLE
Exercise 23: show how to solve consensus in a synchronous distributed system with at most f crash failures

### DISCUSSION THREAD

**Discussion by dag7dev** [sapienzastudentsnetwork/distributed-systems#48] [Exercises]
**Exercise 23:** Show how to solve consensus in a synchronous distributed system with at most f crash failures.

**Comment by susannacifani** [comment]
Here's the solution:

![image](https://github.com/sapienzastudentsnetwork/distributed-system-2023-2024/assets/73530772/6a022f86-9664-42f7-935f-d0646bedba5b)

**Comment by Tommaso-Sgroi** [comment]
To solve consensus in a synchronous distributed system with at most $f$ crash failures and $n$ nodes, each process maintains a set $V$ of values proposed by other processes. Initially, the set contains only the process's value, but then it's updated.
In each round, a process sends to all other processes the values from $V$ that it has not sent before. After $f+1$ rounds, each process decides on the minimum value in $V$.

![image](https://github.com/sapienzastudentsnetwork/distributed-system-2023-2024/assets/62346799/a4f08b03-e496-4386-a229-668eec62b57f)


TOOK FROM: https://cnitarot.github.io/courses/ds_Fall_2016/505_consensus.pdf [page 11]
https://cnitarot.github.io/courses/ds_Fall_2016/index.html#lectures
